experiment_prototype/scan_classes/sequences.py

```diff
@@ -104,203 +78,143 @@ def __init__(self, seqn_keys, sequence_slice_dict, sequence_interface, transmit_
-         (slice_id_1, slice_id_2) and values are of interface_types set up in 
+         (slice_id_1, slice_id_2) and values are of interface_types set up in
          experiment_prototype.
         :param transmit_metadata: metadata from the config file that is useful here.
         """
 
-        # TODO make diagram(s) for pulse combining algorithm
-        # TODO make diagram for pulses that are repeats, showing clearly what intra_pulse_start_time,
-        # and pulse_shift are.
         ScanClassBase.__init__(self, seqn_keys, sequence_slice_dict, sequence_interface,
                                transmit_metadata)
 
-        # TODO: pass clear frequencies to pass to pulses
 
-        # All interfacing at this point is PULSE.
+        txrate = self.transmit_metadata['txrate']
+        txctrfreq = self.transmit_metadata['txctrfreq']
+        main_antenna_count = self.transmit_metadata['main_antenna_count']
+        main_antenna_spacing = self.transmit_metadata['main_antenna_spacing']
+        pulse_ramp_time = self.transmit_metadata['pulse_ramp_time']
+        max_usrp_dac_amplitude = self.transmit_metadata['max_usrp_dac_amplitude']
+        tr_window_time = self.transmit_metadata['tr_window_time']
+
+        self.basic_slice_pulses = {}
+        single_pulse_timing = []
+        for slice_id in self.slice_ids:
+
+            exp_slice = self.slice_dict[slice_id]
+            tx_freq_khz = float(exp_slice['txfreq'])
+            wave_freq = tx_freq_khz - txctrfreq
+            wave_freq_hz = wave_freq * 1000
+
+            if not exp_slice['rxonly']:
+                basic_samples, real_freq = get_samples(txrate, wave_freq_hz,
+                                                   float(exp_slice['pulse_len']) / 1e6,
+                                                   pulse_ramp_time,
+                                                   1.0,
+                                                   exp_slice['iwavetable'],
+                                                   exp_slice['qwavetable'])
+                if real_freq != wave_freq_hz:
+                    errmsg = 'Actual Frequency {} is Not Equal to Intended Wave Freq {}'.format(real_freq,
+                                                                                                wave_freq_hz)
+                    raise ExperimentException(errmsg)  # TODO change to warning? only happens on non-SINE
+
+                # this will zero out non tx antenna samples
+                main_antennas = np.zeros(main_antenna_count, dtype=np.int)
+                temp = np.arange(main_antenna_count)
+                main_antennas[exp_slice['tx_antennas']] = temp[exp_slice['tx_antennas']]
+
+                # convert the beam angles to rads
+                beam_rads = np.pi / 180 * np.array(exp_slice['beam_angle'], dtype=np.float64)
+
+
+                x = ((main_antenna_count / 2.0 - main_antennas) * main_antenna_spacing)
```

Comment:
Can we use the get_phshift function here? I think it's a good idea to use the same function to get the beamform phase offset for both tx and rx in the code. I see a discrepancy here which would lead to an unintended phase offset in the data - the get_phshift function is using (num_antennas - 1)/2 as the reference point and here there is no  - 1. If we need to optimize the get_phshift function too, that's ok, I think it's important that the code is the same though and that's ensured if they both use the same function

Reply:
I've changed the code to match get_phshift. I've mainly focused on tx with the PR, but I think once TX is verified, I'll work on bundling RX code into this as well as debug samples.